### DISCUSSION TITLE
SwitchStore in Preview results in multiple screens

### DISCUSSION THREAD

**Discussion by gsl-anthonymerle** [pointfreeco/swift-composable-architecture#2260] [Q&A] [answered]
### Description

Hi!

So I was working with the `SwitchStore` and wanted to showcase various cases using the XCode's Live Preview feature
There are no crash or important bug _per se_, but in the Preview's Canvas, I see N variations for each use case I have in the preview. With N being the number of cases in my enum State

I tried with a Vanilla SwiftUI `switch` statement in my View, but the behaviour was, as expected, to see only three previews in the canvas

I believe that this behaviour is a direct result of `CaseLet` view using the `IfLetStore` under the hood, which results in multiple `if-else` statements in place of a `switch` statement

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Seeing only 3 preview contents (3 because I have explicitly added 3 previews to my preview provider)

![Capture d’écran 2023-07-04 à 15 24 30](https://github.com/pointfreeco/swift-composable-architecture/assets/107677848/5b25aea4-3bff-40e4-823f-2132d4e93794)

### Actual behavior

I see N preview contents (N being 3 times the number `CaseLet`/`Default` views I have in my `SwitchStore`)

![Capture d’écran 2023-07-04 à 15 19 38](https://github.com/pointfreeco/swift-composable-architecture/assets/107677848/b0611cea-31cf-4be4-b0ce-55e9c1bcb79e)

### Steps to reproduce

Using the following sample, just check the XCode's live preview:
```swift
import ComposableArchitecture
import SwiftUI

struct TestReducer: ReducerProtocol {
    enum State: Equatable {
        case content(String)
        case other(String)
        case nothing
    }

    enum Action: Equatable {}

    var body: some ReducerProtocolOf<Self> {
        EmptyReducer()
    }
}

struct TestView: View {
    let store: StoreOf<TestReducer>

    var body: some View {
        SwitchStore(store) { _ in
            CaseLet(
                state: /TestReducer.State.content,
                action: { (action: TestReducer.Action) in action }
            ) { store in
                Text("Content")
            }

            CaseLet(
                state: /TestReducer.State.other,
                action: { (action: TestReducer.Action) in action }
            ) { store in
                Text("Other")
            }

            CaseLet(
                state: /TestReducer.State.nothing,
                action: { (action: TestReducer.Action) in action }
            ) { store in
                Text("Nothing")
            }
        }
    }
}

struct TestView_Previews: PreviewProvider {
    static var previews: some View {
        TestView(store: .init(
            initialState: .content("Hello, world!"),
            reducer: TestReducer()
        ))
        .previewDisplayName("Content")

        TestView(store: .init(
            initialState: .other("Hello, other world!"),
            reducer: TestReducer()
        ))
        .previewDisplayName("Other")

        TestView(store: .init(
            initialState: .nothing,
            reducer: TestReducer()
        ))
        .previewDisplayName("Loading")
    }
}
```

### The Composable Architecture version information

0.54.0

### Destination operating system

iOS 15 and later

### Xcode version information

Version 14.3.1 (14E300c)

### Swift Compiler version information

```shell
swift-driver version: 1.75.2 Apple Swift version 5.8.1 (swiftlang-5.8.0.124.5 clang-1403.0.22.11.100)
Target: arm64-apple-macosx13.0
```

**Comment by stephencelis** [comment, marked as answer]
`SwitchStore` expects you to explicitly switch over the initial state that is passed along:

```diff
-SwitchStore(store) { _ in
+SwitchStore(store) {
+  switch $0 {
+  case .content:
     CaseLet(
       state: /TestReducer.State.content,
       action: { (action: TestReducer.Action) in action }
     ) { store in
       Text("Content")
     }

+  case .other:
     CaseLet(
       state: /TestReducer.State.other,
       action: { (action: TestReducer.Action) in action }
     ) { store in
       Text("Other")
     }

+  case .nothing:
     CaseLet(
       state: /TestReducer.State.nothing,
       action: { (action: TestReducer.Action) in action }
     ) { store in
       Text("Nothing")
     }
+  }
 }
```

I believe this will get previews to work more as you expect.

Also note that `CaseLet` does not require the `state:` named argument. You can simply do:

```swift
CaseLet(/TestReducer.State.content, /* ... */)
```

Because this isn't a bug in the library I'm going to convert this to a discussion.

**Reply by gsl-anthonymerle** [reply]
Oh! 😳 So I was just using it wrong, my bad 🙏 
Thank you for the explanation @stephencelis 

Can I close the discussion, then ?
